libpromises/crypto.c

```diff
@@ -443,6 +443,14 @@ int EncryptString(char *out, size_t out_length, const char *in, int plainlen,
     if (key == NULL)
         ProgrammingError("EncryptString: session key == NULL");
 
+    size_t max_ciphertext_length = CipherTextLengthMax(CfengineCipher(type), plainlen);
+
+    if(max_ciphertext_length > out_length)
+    {
+        ProgrammingError("EncryptString: output buffer too small: max_ciphertext_length (%ld) > out_length (%ld)",
+                          max_ciphertext_length, out_length);
+    }
+
```

Comment:
This commit should be squashed together with:

* Add utiltiy function CipherTextLengthMax()...
* Pass output buffer size to EncryptString()...